src/hugr/validate/test.rs

```diff
@@ -404,6 +409,7 @@ fn test_ext_edge() -> Result<(), HugrError> {
     Ok(())
 }
 
+#[cfg(feature = "extension_inference")]
```

Comment:
this function should be usable without inference

Reply:
Yeah this is another dead code warning gambit

Reply:
it would be better to add a test that uses it (without needing inference)